
In this Assignment, we shall be working with the Nigerian Car Price Dataset to get the following insights
-the most expensive Automatic car (show car name and model make) with a mileage of less than 50000
-Write a function to bring out the average of each car, which car make has the highest average price? Write a code to show this
-Select the row with a price at the 76th percentile score

# importing libraries and loading dataset

In [29]:
#importing dependencies
import pandas as pd 
import numpy as np 
import matplotlib 
from matplotlib import pyplot as plt 
import seaborn as sb

In [30]:
#loading dataset
cars = pd.read_csv(r"C:\Users\PRAISE\Documents\taskfiles\Nigerian_Car_Prices.csv")

In [31]:
cars.head(5)

Unnamed: 0           Make  Year of manufacture      Condition   Mileage  \
0           0         Toyota               2007.0  Nigerian Used  166418.0   
1           1          Lexus                  NaN            NaN  138024.0   
2           2  Mercedes-Benz               2008.0  Nigerian Used  376807.0   
3           3          Lexus                  NaN            NaN  213362.0   
4           4  Mercedes-Benz                  NaN            NaN  106199.0   

   Engine Size    Fuel Transmission      Price Build  
0       2400.0  Petrol    Automatic  3,120,000   NaN  
1          NaN     NaN    Automatic  5,834,000   NaN  
2       3000.0  Petrol    Automatic  3,640,000   NaN  
3          NaN     NaN    Automatic  3,594,000   NaN  
4          NaN     NaN    Automatic  8,410,000   NaN

# Data cleaning/wrangling

In [32]:
cars.shape

(4095, 10)

In [33]:
# Removing comma from the values in the Price column
cars['Price'] = cars['Price'].str.replace(',','')
cars['Price'].head()

0    3120000
1    5834000
2    3640000
3    3594000
4    8410000
Name: Price, dtype: object

In [34]:
cars.dtypes

Unnamed: 0               int64
Make                    object
Year of manufacture    float64
Condition               object
Mileage                float64
Engine Size            float64
Fuel                    object
Transmission            object
Price                   object
Build                   object
dtype: object

In [35]:
# converting the dtype of Price column from object to integer
cars["Price"] = cars["Price"].astype(int)

In [9]:
#check
#cars.dtypes

In [36]:
cars['Make'].value_counts()#to show the number of cars per type

Toyota           1469
Lexus             464
Mercedes-Benz     436
Honda             428
Ford              197
Hyundai           179
Nissan            173
Kia                90
Land Rover         74
Volkswagen         64
Mazda              63
BMW                63
Acura              56
Infiniti           41
Chevrolet          40
Peugeot            36
Mitsubishi         23
Pontiac            22
Volvo              22
Suzuki             21
Dodge              17
Audi               14
Opel               12
Jaguar             11
Jeep               11
Porsche             9
Renault             8
Chrysler            7
Fiat                6
GMC                 5
Changan             4
Lincoln             4
Skoda               3
Hummer              2
Buick               2
JAC                 2
Citroen             2
Isuzu               2
Mini                2
MG                  2
Rover               1
Bentley             1
Seat                1
SsangYong           1
Cadillac            1
Saturn    

In [26]:
cars.mode()

Unnamed: 0    Make  Year of manufacture      Condition   Mileage  \
0              0  Toyota               2007.0  Nigerian Used  299999.0   
1              1     NaN                  NaN            NaN       NaN   
2              2     NaN                  NaN            NaN       NaN   
3              3     NaN                  NaN            NaN       NaN   
4              4     NaN                  NaN            NaN       NaN   
...          ...     ...                  ...            ...       ...   
4090        4090     NaN                  NaN            NaN       NaN   
4091        4091     NaN                  NaN            NaN       NaN   
4092        4092     NaN                  NaN            NaN       NaN   
4093        4093     NaN                  NaN            NaN       NaN   
4094        4094     NaN                  NaN            NaN       NaN   

      Engine Size    Fuel Transmission      Price Build  
0          3500.0  Petrol    Automatic  2625000.0   SUV  
1             NaN     NaN          NaN        NaN   NaN  
2             NaN     NaN          NaN        NaN   NaN  
3             NaN     NaN          NaN        NaN   NaN  
4             NaN     NaN          NaN        NaN   NaN  
...           ...     ...          ...        ...   ...  
4090          NaN     NaN          NaN        NaN   NaN  
4091          NaN     NaN          NaN        NaN   NaN  
4092          NaN     NaN          NaN        NaN   NaN  
4093          NaN     NaN          NaN        NaN   NaN  
4094          NaN     NaN          NaN        NaN   NaN  

[4095 rows x 10 columns]

In [37]:
#checking for missing values
cars.isna().sum()

Unnamed: 0                0
Make                      0
Year of manufacture     478
Condition               479
Mileage                  71
Engine Size             511
Fuel                    488
Transmission             20
Price                     0
Build                  2968
dtype: int64

checking for missing values, we find out that there exist in columns like year of manufacture with 478,condtion with 469,mileage with 71 etc.iwill be filling them up with the mode.

In [40]:
cars=cars.fillna(cars.mode().iloc[0])#to fill our Null values using mode
#check
cars.isna().sum()

Unnamed: 0             0
Make                   0
Year of manufacture    0
Condition              0
Mileage                0
Engine Size            0
Fuel                   0
Transmission           0
Price                  0
Build                  0
dtype: int64

In [42]:
# Describing the dataset to have more insight
cars.describe()

Unnamed: 0  Year of manufacture       Mileage   Engine Size  \
count  4095.000000          4095.000000  4.095000e+03    4095.00000   
mean   2047.000000          2007.793651  1.845703e+05    3303.05641   
std    1182.269005             4.051592  2.096479e+05    7197.73812   
min       0.000000          1992.000000  1.000000e+00       3.00000   
25%    1023.500000          2005.000000  1.025985e+05    2400.00000   
50%    2047.000000          2007.000000  1.637560e+05    3000.00000   
75%    3070.500000          2010.000000  2.365910e+05    3500.00000   
max    4094.000000          2021.000000  9.976050e+06  371000.00000   

              Price  
count  4.095000e+03  
mean   4.271288e+06  
std    4.900064e+06  
min    4.580000e+05  
25%    1.872000e+06  
50%    2.940000e+06  
75%    4.725000e+06  
max    5.880000e+07

Q1: What is the most expensive Automatic car (show car name and model make) with a mileage of less than 50000

In [44]:
cars.groupby(['Make','Build'])[['Make','Build','Price']].max().max()
#the most expensive car is a Volve make, with SUV build and it's price is 58800000(fifty eight million,eight hundred thousand)


Make        Volvo
Build         SUV
Price    58800000
dtype: object

Q2: Write a function to bring out the average of each car, which car make has the highest average price? Write a code to show this



In [45]:
 #Average Of Each Car
cars.groupby(['Make'])['Price'].mean()


Make
Acura            3.414470e+06
Audi             6.090536e+06
BMW              3.785456e+06
Bentley          1.689500e+07
Buick            4.770000e+06
Cadillac         6.730000e+06
Changan          1.222500e+07
Chevrolet        4.473020e+06
Chrysler         2.797500e+06
Citroen          1.260000e+06
Daihatsu         9.450000e+05
Dodge            4.898382e+06
Fiat             6.945000e+06
Ford             3.711745e+06
Foton            4.938000e+06
GMC              8.784600e+06
Honda            2.527255e+06
Hummer           6.300000e+06
Hyundai          3.873316e+06
Infiniti         2.997098e+06
Isuzu            2.048400e+06
JAC              7.290000e+06
Jaguar           4.330682e+06
Jeep             9.580545e+06
Kia              3.511190e+06
Land Rover       1.021207e+07
Lexus            5.032669e+06
Lincoln          7.601000e+06
MG               1.575000e+06
Mazda            2.760742e+06
Mercedes-Benz    6.270841e+06
Mini             6.300000e+06
Mitsubishi       1.920609e+06
Nissa

In [46]:
#car with highest average price
cars.groupby(['Make','Build'])['Price'].mean().reset_index().max().reset_index()

index           0
0   Make       Volvo
1  Build         SUV
2  Price  17850000.0

Select the row with a price at the 76th percentile score

In [47]:
cars.quantile([0.76], axis=0)

Unnamed: 0  Year of manufacture    Mileage  Engine Size      Price
0.76     3111.44               2010.0  241519.32       3500.0  4770000.0